src/table/components/head/styles.ts

```diff
@@ -49,27 +54,58 @@ export const VisuallyHidden = styled('span')({
   width: 1,
 });
 
-export const StyledHeadCellContent = styled(Box)(({ theme }) => ({
+export const StyledHeadCellContent = styled(Box, {
+  shouldForwardProp: (prop: string) => prop !== 'isLocked',
+})(({ theme, isLocked }) => ({
   width: '100%',
-  display: 'flex',
-  justifyContent: 'space-between',
+  display: 'grid',
   flexDirection: 'inherit',
-  alignItems: 'flex-end',
+  gridAutoFlow: 'dense',
   gap: theme.spacing(0.5),
```

Comment:
do you still need this? I think in general since you are introducing a lot of new css, make sure to clean out stuff that is not needed

Reply:
Yes, that is because of `grid-column` property makes items in separate lines and making them dense -> puts them back into one line